libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -529,57 +569,217 @@ where
     fn add_inbound_connection(
         &mut self,
         connection: Box<dyn Connection>,
-    ) -> Result<String, ConnectionManagerError> {
+        reply_sender: Sender<Result<(), ConnectionManagerError>>,
+        internal_sender: Sender<CmMessage>,
+        auth_pool: &AuthorizationPool,
+        subscribers: &mut SubscriberMap,
+    ) {
         let endpoint = connection.remote_endpoint();
-
         let id = Uuid::new_v4().to_string();
 
+        if endpoint.contains("inproc") {
```

Comment:
Why is inproc special?

Reply:
Inproc was previously special, in that it didn't require auth (it's compiled in, after all).  This has just shifted to let the connection manager decide, versus the auth layer, which should have to be concerned about what type of connection it is.

Reply:
It probably would be better to add a feature where a connection could be indicate that it is authorizable, but that would require some API changes at the transport level.  Not hard - the default implementation of the function could return true.